interview-questions.md

```diff
@@ -7,53 +7,54 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-  Researched answer:
+  Researched answer: Now that i went deeper on MVC i learn more then what i answer and what i just learn is the pattern that MVC uses archiectural. each of them has it's job like model handles the data and logic, controllers handles the users interface and applications, and view handles graphical users objects and presentation.
 
 
 
 2. What is a relational database? Are there other kinds of databases?
 
-  Your answer:
+  Your answer: From what i know about RD that Relational database includes tables containing rows and columns. Also, from what i learn while resarching that RB (Relational Databases) are also called Relational Database Management Systems.
 
-  Researched answer:
+  Researched answer: so i did my more reserach on relational database and i got a better undertsanding on it so what i learn each row in the table is a record with a unique ID called the key and columns of the table hold attributes of the data. for example;  each record includes a customer’s name, address, shipping and billing information, phone number, and other contact information.
 
 
 
 3. What is object-relational mapping?
 
-  Your answer:
+  Your answer: From what i learn in class ORM (object-relational mapping) is a tool that allows us to write SQL queries using rails. and i know Object part is the one you use with your programming language, Relational part is a Relational Database Manager System and finally the Mapping part is where you do a bridge between your objects and your tables that is how i think of ORM (object-relational mapping)
 
-  Researched answer:
+  Researched answer: object-relational mapping is very important becuase it allows ruby to call methods and recives information from the object. this is what intrest me that they're a lot of different type of ORM languages and it is Django ORM, Dapper ORM, JOOQ ORM and  SQL Alchemy.
 
 
 
 4. What is a gem?
 
-  Your answer:
+  Your answer: i don't know much about it i need to do more reserach in it but what i know from class gems package up Ruby code so it is easy to share with others. I also remember gem is like a library that can be installed in your server in projects.
 
-  Researched answer:
+  Researched answer: so i reserached more about gem and what i've learned online that each gem has a name, version and plateform and ruby was first appered in 1995 and the developer who designed was Yukihiro Matsumoto and gem was appeared in 2003 honestly i thought that was really cool.
 
 
 
 5. What are primary keys? Why are they important?
 
-  Your answer:
+  Your answer:so far from what i know in class that the primary key is are columns that contain valuse that uniquely to identifiy each row in the table. 
 
-  Researched answer:
+  Researched answer: So i did a lot of reserach for primary key and what i've learn that the primary key must have Optim to insert, update, restore or delete data from the database table.
```

Comment:
good and close! Primary keys are what let us associate two tables in a database with each other.